cardano-api/src/Cardano/Api/LocalStateQuery.hs

```diff
@@ -0,0 +1,188 @@
+import           Data.Functor.Contravariant (contramap)
+import           Data.Text (Text)
+import qualified Data.Text as Text
+import           Data.Void (Void)
+
+import           Network.Mux (MuxTrace, WithMuxBearer)
+
+import           Ouroboros.Consensus.Block (BlockProtocol)
+import           Ouroboros.Consensus.Cardano (Protocol (..), protocolInfo)
+import           Ouroboros.Consensus.Config (TopLevelConfig (..))
+import           Ouroboros.Consensus.Ledger.Abstract (Query)
+import           Ouroboros.Consensus.Network.NodeToClient
+import           Ouroboros.Consensus.Node.NetworkProtocolVersion
+                  (nodeToClientProtocolVersion, supportedNodeToClientVersions)
+import           Ouroboros.Consensus.Node.ProtocolInfo (ProtocolInfo(..))
+import           Ouroboros.Consensus.Node.Run (RunNode, nodeNetworkMagic)
+import           Ouroboros.Consensus.Shelley.Ledger
+import           Ouroboros.Consensus.Shelley.Protocol.Crypto (TPraosCrypto)
+
+import           Ouroboros.Network.Block (Point)
+import           Ouroboros.Network.Mux
+                   ( AppType(..), OuroborosApplication(..),
+                     MuxPeer(..), RunMiniProtocol(..))
+import           Ouroboros.Network.NodeToClient (NetworkConnectTracers (..), TraceSendRecv)
+import qualified Ouroboros.Network.NodeToClient as NtC
+import           Ouroboros.Network.Protocol.LocalStateQuery.Client
+                   (ClientStAcquired (..), ClientStAcquiring (..), ClientStIdle (..),
+                    ClientStQuerying (..), LocalStateQueryClient(..), localStateQueryClientPeer)
+import           Ouroboros.Network.Protocol.LocalStateQuery.Type (AcquireFailure (..))
+
+import qualified Shelley.Spec.Ledger.PParams as Ledger (PParams)
+
+-- | Query the current protocol parameters from a Shelley node via the local
+-- state query protocol.
+queryPParamsFromLocalState
+  :: TPraosCrypto c
+  => Protocol (ShelleyBlock c) (BlockProtocol (ShelleyBlock c))
+  -> SocketPath
+  -> Point (ShelleyBlock c)
+  -> ExceptT AcquireFailure IO Ledger.PParams
+queryPParamsFromLocalState ptcl socketPath point = do
+  let pointAndQuery = (point, GetCurrentPParams)
+  pParamsVar <- liftIO newEmptyTMVarM
+  liftIO $ queryNodeLocalState
+    pParamsVar
+    nullTracer
+    ptcl
+    socketPath
+    pointAndQuery
+  newExceptT $ atomically $ takeTMVar pParamsVar
+
+-- | Establish a connection to a Shelley node and execute the provided query
+-- via the local state query protocol.
+queryNodeLocalState
+  :: forall blk result. RunNode blk
+  => StrictTMVar IO (Either AcquireFailure result)
+  -> Trace IO Text
+  -> Protocol blk (BlockProtocol blk)
+  -> SocketPath
+  -> (Point blk, Query blk result)
+  -> IO ()
+queryNodeLocalState resultVar trce ptcl (SocketFile socketPath) pointAndQuery = do
+    logInfo trce $ "queryNodeLocalState: Connecting to node via " <> textShow socketPath
+    let ProtocolInfo{pInfoConfig} = protocolInfo ptcl
+    NtC.withIOManager $ \iocp -> do
```

Comment:
Had to scroll up to figure out what `NtC` is. Might be better to use `NodeToClient`. Yes, its more characters but far more readable.